webapp/lib/controller.dart

```diff
@@ -474,6 +474,19 @@ void initUI() {
         ..addAll(added)
         ..addAll(modified);
       view.conversationListSelectView.updateConversationLists(shards);
+
+      // Read any conversation shards from the URL
+      String urlConversationListRoot = view.urlView.getPageUrlConversationList();
```

Comment:
Do you think this might be better as a helper function?

Reply:
I'd rather not - it's a bit of a complicated piece of logic, and this is the only place where it needs to be computed, so I'd prefer to keep it in context, for now at least